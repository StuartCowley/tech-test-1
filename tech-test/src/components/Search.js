import React, { useState } from "react";
import PropTypes from "prop-types";
import getImages from "../requests/getImages";
import "../styles/Search.css";

const Search = ({ setSearchResults }) => {
  const [searchValue, setSearchValue] = useState();

  const handleSubmit = async (event) => {
    event.preventDefault();
    setSearchResults(await getImages(searchValue));
  };

  return (
    <div className="Search">
      <form className="seach-form" onSubmit={handleSubmit}>
        <input 
          className="search-input" 
          placeholder="Search Images"
          type="text"
          onChange={(e) => setSearchValue(e.target.value)}
        />
        <button
          className="search-button"
          type="submit"
        >Launch!</button>
      </form>
    </div>
  );
};

export default Search;

Search.propTypes ={
  setSearchResults: PropTypes.func,
};