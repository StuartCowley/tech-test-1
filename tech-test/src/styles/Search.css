.Search {
  padding-bottom: 50px;
}

.form {
  display: flex;
  flex-direction: row;
}

.search-input {
  width: 300px;
  height: 50px;
  border-radius: 11px;
  font-size: 20px;
  padding: 0 10px;
}

.search-button {
  width: 100px;
  height: 50px;
  border-radius: 11px;
  padding: 0 10px;
  font-size: 20px;
  transition-duration: 0.4s;
}

.search-button:hover {
  background-color: green;
  color: white;
}