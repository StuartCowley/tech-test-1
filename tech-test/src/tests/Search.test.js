import { render, screen } from "@testing-library/react";
import React from "react";
import Search from "../components/Search";

describe("Search", () => {

  const validProps = {
    setSearchResults: () => {}
  };

  it("renders Search component correctly", () => {
    const { asFragment } = render(
    <Search 
      setSearchResults={validProps.setSearchResults} 
    />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders button element", () => {
    render(
      <Search
        setSearchResults={validProps.setSearchResults}
      />
    );
    const launchButton = screen.getByText("Launch!")
    expect(launchButton).toHaveClass("search-button");
    expect(launchButton).toBeInTheDocument();
  })
});