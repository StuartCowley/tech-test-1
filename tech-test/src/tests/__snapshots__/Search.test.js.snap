// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders Search component correctly 1`] = `
<DocumentFragment>
  <div
    class="Search"
  >
    <form
      class="seach-form"
    >
      <input
        class="search-input"
        placeholder="Search Images"
        type="text"
      />
      <button
        class="search-button"
        type="submit"
      >
        Launch!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
